src/Databases/DataLake/DatabaseDataLake.cpp

```diff
@@ -490,27 +538,40 @@ void registerDatabaseDataLake(DatabaseFactory & factory)
         const auto * database_engine_define = args.create_query.storage;
         const auto & database_engine_name = args.engine_name;
 
+        DatabaseDataLakeSettings database_settings;
+        if (database_engine_define->settings)
+            database_settings.loadFromQuery(*database_engine_define);
+
+        auto catalog_type = database_settings[DB::DatabaseDataLakeSetting::catalog_type].value;
         const ASTFunction * function_define = database_engine_define->engine;
-        if (!function_define->arguments)
+        ASTs engine_args;
+        if (!function_define->arguments && catalog_type != DatabaseDataLakeCatalogType::GLUE)
+        {
             throw Exception(ErrorCodes::BAD_ARGUMENTS, "Engine `{}` must have arguments", database_engine_name);
+        }
 
-        ASTs & engine_args = function_define->arguments->children;
-        if (engine_args.empty())
-            throw Exception(ErrorCodes::BAD_ARGUMENTS, "Engine `{}` must have arguments", database_engine_name);
+        if (function_define->arguments)
+        {
+            engine_args = function_define->arguments->children;
+            if (engine_args.empty() && catalog_type != DatabaseDataLakeCatalogType::GLUE)
```

Comment:
Since it appears several times, I'd make is a separate variable 
```
const bool requires_arguments = catalog_type != DatabaseDataLakeCatalogType::GLUE;
```
and a comment above it as I mentioned in a previous comment